### ISSUE TITLE
to_thread.run_sync() doesn't work with the tornado event loop

### ISSUE THREAD

**Issue by mwakaba2** [agronholm/anyio#264]
Hi there!

[jupyter_server](https://github.com/jupyter-server/jupyter_server)  uses the anyio v2 `run_sync_in_worker_thread` function , and I noticed a possible bug after upgrading to anyio v3. 

This is the error I see when using `to_thread.run_sync()`:
```python
HTTPServerRequest(protocol='http', host='127.0.0.1:38403', method='GET', uri='/a%40b/api/contents/Directory%20with%20spaces%20in', version='HTTP/1.1', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/anyio/_backends/_asyncio.py", line 697, in run_sync_in_worker_thread
    work_queue = _threadpool_work_queue.get()
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/anyio/lowlevel.py", line 120, in get
    raise LookupError(f'Run variable "{self._name}" has no value and no default set')
LookupError: Run variable "_threadpool_work_queue" has no value and no default set

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/tornado/web.py", line 1704, in _execute
    result = await result
  File "/home/runner/work/jupyter_server/jupyter_server/jupyter_server/services/contents/handlers.py", line 111, in get
    path=path, type=type, format=format, content=content,
  File "/home/runner/work/jupyter_server/jupyter_server/jupyter_server/utils.py", line 176, in ensure_async
    result = await obj
  File "/home/runner/work/jupyter_server/jupyter_server/jupyter_server/services/contents/filemanager.py", line 692, in get
    model = await self._dir_model(path, content=content)
  File "/home/runner/work/jupyter_server/jupyter_server/jupyter_server/services/contents/filemanager.py", line 574, in _dir_model
    dir_contents = await run_sync_in_worker_thread(os.listdir, os_dir)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/anyio/to_thread.py", line 37, in run_sync_in_worker_thread
    return await run_sync(func, *args, cancellable=cancellable, limiter=limiter)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/anyio/to_thread.py", line 29, in run_sync
    limiter=limiter)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/anyio/_backends/_asyncio.py", line 707, in run_sync_in_worker_thread
    find_root_task().add_done_callback(_loop_shutdown_callback)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/anyio/_backends/_asyncio.py", line 56, in find_root_task
Error:     raise RuntimeError('Cannot find root task for setting cleanup callback')
```
It looks like the `find_root_task` function is failing because the callback is a tornado.ioloop and not an uvloop.loop.

Jupyter server is using the tornado event loop.
For more context, this is the github issue: https://github.com/jupyter-server/jupyter_server/issues/487

**Possible Solution**

I was able to fix it by updating the following code, but I'm not sure if this is a proper fix.  
[src/anyio/_backends/_asyncio.py line 50](https://github.com/agronholm/anyio/pull/227/files#diff-202c3fc4551e5a287df488a9bd96a25dcc1405ec135d5c7aceb6ef4c02755341R50) 

```python
if cb is _run_until_complete_cb or cb.__module__ in ('uvloop.loop', 'tornado.ioloop')
```

**Comment by agronholm** [comment]
Then I would have to list every asyncio loop implementation under the blue sky. I wonder if stack traversal is a preferable alternative, or if that would even work.

**Comment by mwakaba2** [comment]
yeah that's not ideal. Tornado.ioloop is a wrapper around the asyncio event loop. 
I'm curious to learn what this check is doing for the callback.

```python
if cb is _run_until_complete_cb or cb.__module__ == 'uvloop.loop':
```

**Comment by agronholm** [comment]
The point was to find the "root" task or future so that the cleanup callback could be hooked into it, to ensure that the thread pool is shut down along with the event loop (or a `run_until_complete()` call.

**Comment by agronholm** [comment]
This will likely require a nontrivial fix.

**Comment by agronholm** [comment]
We discussed this over at the Gitter room and the only viable generic solution seems to be to make the workers daemonic and move the cleanup to an `atexit()` handler.

**Comment by agronholm** [comment]
I will try to resolve this over the coming weekend.

**Comment by agronholm** [comment]
@mwakaba2 Please test PR #268 as a potential fix for this issue.

**Comment by mwakaba2** [comment]
@agronholm I no longer see the error after applying your PR. Thanks so much for the quick refactor! 🙂

**Comment by mwakaba2** [comment]
@agronholm I reviewed your code changes, and it seems fine, but it's probably better to get a code review from the collaborators at some point. I might look into performance testing, comparing anyio v2 & v3 with the Jupyter Server if that makes sense. Please let me know if I can help with any other tests.

**Comment by agronholm** [comment]
I've released v3.0.1 with this fix now.
